### PR TITLE
ROX-15458: Forward Node-Scanning-ACKs from Central to Compliance

### PR DESCRIPTION
**Pull request by vikin91**
## Description

This PR builds on top of the previous one (#6155). Here, Sensor forwards the ACK message it received to Compliance.
I also changed the type of proto message Sensor<->Compliance to match the Central<->Sensor one.

In the last follow-up: We will implement the behavior of Compliance for when it does NOT receive an ACK for a defined duration. This follow-up is partially blocked by #5999 (due to refactoring and potential many merge-conflicts).    

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added

### N/A
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed

- CI
- Existing unit tests for NodeInventoryHandler
- Manual deployment into cluster and looking at the debug logs (see below)

#### Central

```
central-6bbf5ffb6c-4lpt2 sensor/service/pipeline/nodeinventory: 2023/05/25 15:38:34.310449 pipeline.go:92: Info: Scanned node inventory remote/piotr-work2-d7t26-master-2.c.srox-temp-dev-test.internal (id: f2b2d5d6-f383-4b98-ac80-82fc88553ece) with 511 components
central-6bbf5ffb6c-4lpt2 sensor/service/pipeline/nodeinventory: 2023/05/25 15:38:36.128710 pipeline.go:92: Info: Scanned node inventory remote/piotr-work2-d7t26-master-1.c.srox-temp-dev-test.internal (id: fdaae223-31b5-481f-abca-9602419b9ad6) with 511 components
central-6bbf5ffb6c-4lpt2 sensor/service/pipeline/nodeinventory: 2023/05/25 15:38:37.836517 pipeline.go:92: Info: Scanned node inventory remote/piotr-work2-d7t26-master-0.c.srox-temp-dev-test.internal (id: 91ae5d20-40d7-4649-bb45-2ad69fa006e7) with 511 components
central-6bbf5ffb6c-4lpt2 sensor/service/pipeline/nodeinventory: 2023/05/25 15:38:38.598145 pipeline.go:92: Info: Scanned node inventory remote/piotr-work2-d7t26-worker-c-bj6lb.c.srox-temp-dev-test.internal (id: 20ecf815-c939-4a39-b80f-50b89a28c885) with 511 components
central-6bbf5ffb6c-4lpt2 sensor/service/pipeline/nodeinventory: 2023/05/25 15:38:38.917023 pipeline.go:92: Info: Scanned node inventory remote/piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal (id: ca96656a-368b-4717-a3cb-17d89bcf35d5) with 511 components
```

#### Sensor

```
common/compliance: 2023/05/25 15:35:31.795021 node_inventory_handler_impl.go:120: Debug: Forwarding node-scanning-ACK message from Central to Compliance: {ack:<>  piotr-work2-d7t26-worker-c-bj6lb.c.srox-temp-dev-test.internal false}
common/compliance: 2023/05/25 15:35:33.040008 node_inventory_handler_impl.go:142: Debug: Mapping NodeInventory name 'piotr-work2-d7t26-worker-d-zr9d4.c.srox-temp-dev-test.internal' to Node ID '29b63bac-4a52-4764-a80e-f7b6aa6344e9'
common/compliance: 2023/05/25 15:35:33.412257 node_inventory_handler_impl.go:86: Debug: Received node-scanning-ACK message: cluster_id:"4f9b2e5a-5812-45b4-965b-736753b99754" node_name:"piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
common/compliance: 2023/05/25 15:35:33.412385 node_inventory_handler_impl.go:120: Debug: Forwarding node-scanning-ACK message from Central to Compliance: {ack:<>  piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal false}
common/compliance: 2023/05/25 15:35:33.708405 node_inventory_handler_impl.go:86: Debug: Received node-scanning-ACK message: cluster_id:"4f9b2e5a-5812-45b4-965b-736753b99754" node_name:"piotr-work2-d7t26-worker-c-bj6lb.c.srox-temp-dev-test.internal"
common/compliance: 2023/05/25 15:35:33.708531 node_inventory_handler_impl.go:120: Debug: Forwarding node-scanning-ACK message from Central to Compliance: {ack:<>  piotr-work2-d7t26-worker-c-bj6lb.c.srox-temp-dev-test.internal false}
common/compliance: 2023/05/25 15:35:33.967054 node_inventory_handler_impl.go:142: Debug: Mapping NodeInventory name 'piotr-work2-d7t26-master-2.c.srox-temp-dev-test.internal' to Node ID 'f2b2d5d6-f383-4b98-ac80-82fc88553ece'
common/compliance: 2023/05/25 15:35:34.440583 node_inventory_handler_impl.go:142: Debug: Mapping NodeInventory name 'piotr-work2-d7t26-master-1.c.srox-temp-dev-test.internal' to Node ID 'fdaae223-31b5-481f-abca-9602419b9ad6'
common/compliance: 2023/05/25 15:35:35.258565 node_inventory_handler_impl.go:86: Debug: Received node-scanning-ACK message: cluster_id:"4f9b2e5a-5812-45b4-965b-736753b99754" node_name:"piotr-work2-d7t26-worker-d-zr9d4.c.srox-temp-dev-test.internal"
common/compliance: 2023/05/25 15:35:35.258699 node_inventory_handler_impl.go:120: Debug: Forwarding node-scanning-ACK message from Central to Compliance: {ack:<>  piotr-work2-d7t26-worker-d-zr9d4.c.srox-temp-dev-test.internal false}
common/compliance: 2023/05/25 15:35:35.582610 node_inventory_handler_impl.go:142: Debug: Mapping NodeInventory name 'piotr-work2-d7t26-master-1.c.srox-temp-dev-test.internal' to Node ID 'fdaae223-31b5-481f-abca-9602419b9ad6'
common/compliance: 2023/05/25 15:35:35.905854 node_inventory_handler_impl.go:86: Debug: Received node-scanning-ACK message: cluster_id:"4f9b2e5a-5812-45b4-965b-736753b99754" node_name:"piotr-work2-d7t26-master-2.c.srox-temp-dev-test.internal"
common/compliance: 2023/05/25 15:35:35.906087 node_inventory_handler_impl.go:120: Debug: Forwarding node-scanning-ACK message from Central to Compliance: {ack:<>  piotr-work2-d7t26-master-2.c.srox-temp-dev-test.internal false}
```

#### Selected Compliance container

```
main: 2023/05/25 15:38:37.098253 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:38:37.101368 intervals.go:73: Info: Next node scan in 2.54236246s
main: 2023/05/25 15:38:39.038046 main.go:107: Info: Received node-scanning-ACK from Sensor
main: 2023/05/25 15:38:39.644689 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:38:39.649549 intervals.go:73: Info: Next node scan in 15.604298528s
main: 2023/05/25 15:38:41.493610 main.go:107: Info: Received node-scanning-ACK from Sensor
main: 2023/05/25 15:38:55.254492 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:38:55.258863 intervals.go:73: Info: Next node scan in 10.342031201s
main: 2023/05/25 15:38:57.155512 main.go:107: Info: Received node-scanning-ACK from Sensor
main: 2023/05/25 15:39:05.601994 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:39:05.606906 intervals.go:73: Info: Next node scan in 10.542946994s
main: 2023/05/25 15:39:07.508844 main.go:107: Info: Received node-scanning-ACK from Sensor
main: 2023/05/25 15:39:16.149938 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:39:16.154394 intervals.go:73: Info: Next node scan in 9.941748345s
main: 2023/05/25 15:39:17.981235 main.go:107: Info: Received node-scanning-ACK from Sensor
main: 2023/05/25 15:39:26.096925 main.go:191: Info: Scanning node "piotr-work2-d7t26-worker-b-z7gpb.c.srox-temp-dev-test.internal"
collection/intervals: 2023/05/25 15:39:26.100482 intervals.go:73: Info: Next node scan in 9.365752507s
main: 2023/05/25 15:39:27.951535 main.go:107: Info: Received node-scanning-ACK from Sensor
```

### CODE REVIEW

**Review by jschnath — APPROVED** [review]
Looks fine to me, I just have some questions, see comments

File: sensor/common/compliance/node_inventory_handler_impl.go [hunk 1]
```diff
@@ -129,20 +150,18 @@ func (c *nodeInventoryHandlerImpl) nodeInventoryHandlingLoop(toCentral chan *cen
...
+			Msg: &sensor.MsgToCompliance_Ack{
+				Ack: &sensor.MsgToCompliance_NodeInventoryACK{
+					Action: action,
 				},
 			},
 		},
-		Hostname:  inventory.GetNodeName(),
-		Broadcast: false,
+		Hostname:  nodeName,
+		Broadcast: nodeName == "",
```
[the remark this answers is not in the document]
Comment (vikin91) [reply sensor/common/compliance/node_inventory_handler_impl.go:164]: If we do not have node name, then we do not have an address where to send the message. So `broadcast == true` when `nodeName == ""`. Makes sense?

File: proto/internalapi/sensor/compliance_iservice.proto [hunk 2]
```diff
@@ -51,20 +51,19 @@ message MsgToCompliance {
...
-        string          node_id    = 1;
-    }
-    message NodeInventoryAck {
-        string          node_id    = 1;
+    message NodeInventoryACK {
+        Action  action = 1;
+        enum Action {
+            ACK = 0;
+            NACK = 1;
+        }
```
[the remark this answers is not in the document]
Comment (vikin91) [reply proto/internalapi/sensor/compliance_iservice.proto:59]: I asked myself why we need it at all and found that is is not used - we do not read that in Compliance, do we?

File: sensor/common/compliance/node_inventory_handler_impl.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jschnath) [reply sensor/common/compliance/node_inventory_handler_impl.go:164]: Okay, makes sense

File: proto/internalapi/sensor/compliance_iservice.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (jschnath) [reply proto/internalapi/sensor/compliance_iservice.proto:59]: I double checked this and found no usages either, I guess it was redundant

**Comment by vikin91** [comment]
Many thanks for the review! I will merge this at the beginning of next week

**Pull request merged**